### PR TITLE
[Comments] Website overlay example

### PR DESCRIPTION
**Pull request by CTNicholas**
This example started as a [Strapi CMS demo](https://github.com/liveblocks/strapi-demo/). It has the comments and cursors, but I've removed the editable text.

**TODO** 
- [x] Thumbnail for README @pierrelevaillant 

**TODO AFTER MERGE**
- [ ] Deploy @CTNicholas 
- [ ] Add to website examples @CTNicholas

### CODE REVIEW

File: examples/nextjs-comments-website-overlay/src/components/comments/Sidebar.module.css [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  overflow: hidden;
+}
+
+.sidebarThread[data-thread-resolved] {
+  opacity: 0.65;
+}
+
+/*.sidebarThread[data-thread-resolved]:hover {*/
+/*  opacity: 0.8;*/
+/*}*/
```
Comment (marcbouchenoire) [inline examples/nextjs-comments-website-overlay/src/components/comments/Sidebar.module.css:90]: To be removed.

File: examples/nextjs-comments-website-overlay/src/styles/comments.css [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+.lb-tooltip,
+.lb-dropdown-item,
+.lb-composer-suggestions {
+  z-index: var(--z-above-comments-sidemenu);
+}
+
+/* Temporary, until we update comments */
+[data-radix-popper-content-wrapper] {
+  z-index: var(--z-above-comments-sidemenu) !important;
+}
```
Comment (marcbouchenoire) [inline examples/nextjs-comments-website-overlay/src/styles/comments.css:24]: `.lb-portal` can be used now.

File: examples/nextjs-comments-website-overlay/src/components/cursors/Cursor.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      animate={coords}
+      transition={{
+        type: "spring",
+        damping: 20,
+        mass: 0.4,
+        stiffness: 400,
+      }}
+    >
+      <svg
+        className="h-4 w-auto lg:h-5"
```
Comment (marcbouchenoire) [inline examples/nextjs-comments-website-overlay/src/components/cursors/Cursor.tsx:44]: Tailwind remnants that need to be removed or replaced.

**Review by marcbouchenoire — CHANGES_REQUESTED** [review]
Great work! I added a few small comments, and I tried to run it locally but I'm missing some pieces:


Also I'm not entirely sure about the name, it's a bit long. Would `nextjs-comments-overlay` sound like something else?

**Comment by CTNicholas** [comment]
Thanks for the review @marcbouchenoire! I've made all the changes 👍

There was an error in an icon file, but I've fixed it, so you should see the `+` icon now.

**Comment by marcbouchenoire** [comment]
Nice! Some comments:
- You could add `autoFocus` to the `Composer` displayed to create new threads because it feels a bit weird to have to re-click
- The popup to create new threads can close itself maybe too easily, like here when clicking the emoji button (which makes the editor unfocused)

https://github.com/liveblocks/liveblocks/assets/6959425/96e333e1-6d79-47f4-bd4a-49093fc5f094

File: examples/nextjs-comments-overlay/src/styles/comments.css [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+.lb-tooltip,
+.lb-dropdown-item,
+.lb-composer-suggestions {
+  z-index: var(--z-above-comments-sidemenu);
+}
+
+.lb-portal {
+  z-index: var(--z-above-comments-sidemenu) !important;
```
Comment (marcbouchenoire) [inline examples/nextjs-comments-overlay/src/styles/comments.css:22]: `!important` won't be necessary when the CSS improvements land in the next release.

**Comment by CTNicholas** [comment]
All done! 👍

**Review by marcbouchenoire — APPROVED** [review]
🙌

File: examples/nextjs-comments-overlay/src/components/comments/Comments.tsx [hunk 5]
```diff
@@ -3,16 +3,27 @@
...
   return (
-    <ClientSideSuspense fallback={null}>
-      {() => (
-        <>
-          <Toolbar />
-          <CommentsOverlay />
-        </>
-      )}
-    </ClientSideSuspense>
+    /* @ts-ignore */
```
Comment (marcbouchenoire) [inline examples/nextjs-comments-overlay/src/components/comments/Comments.tsx:11]: What was the TS error?

File: examples/nextjs-comments-overlay/tsconfig.json [hunk 6]
```diff
@@ -9,7 +9,7 @@
     "noEmit": true,
     "esModuleInterop": true,
     "module": "esnext",
-    "moduleResolution": "bundler",
```
[the remark this answers is not in the document]
Comment (CTNicholas) [reply examples/nextjs-comments-overlay/tsconfig.json]: Yep, same error we saw before

File: examples/nextjs-comments-overlay/src/components/comments/Comments.tsx (hunk 5 above)
Comment (CTNicholas) [reply examples/nextjs-comments-overlay/src/components/comments/Comments.tsx:11]: `'ErrorBoundary' cannot be used as a JSX component`

Didn't see a way around it, no solutions online worked.

**Pull request merged**